src/NexusMods.App.UI/Controls/TreeDataGrid/TreeDataGridAdapter.cs

```diff
@@ -0,0 +1,148 @@
+using Avalonia.Controls;
+using Avalonia.Controls.Models.TreeDataGrid;
+using Avalonia.Controls.Selection;
+using DynamicData;
+using NexusMods.App.UI.Extensions;
+using ObservableCollections;
+using R3;
+
+namespace NexusMods.App.UI.Controls;
+
+public abstract class TreeDataGridAdapter<TModel> : ReactiveR3Object
```

Comment:
Since this is new stuff, we should add comments to the new classes, to give some idea on their purpose and when they should be used.